wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/RemoveClients/RemoveClientsViewController.swift

```diff
@@ -138,19 +122,33 @@ final class RemoveClientsViewController: UIViewController,
         self.navigationController?.presentingViewController?.dismiss(animated: true, completion: nil)
     }
 
-    func removeUserClient(_ userClient: UserClient) {
-        userClientToDelete = userClient
-        present(requestPasswordController.alertController, animated: true)
+    func removeUserClient(_ userClient: UserClient) async {
+        if let password = await presentRequestPasswordController() {
+            await removeUserClient(userClient, password: password)
+        }
     }
 
     // MARK: - Helpers
 
-    private func removeUserClient(password: String) async {
-        guard let userClientToDelete = userClientToDelete else { return }
+    private func presentRequestPasswordController() async -> String? {
+        await withCheckedContinuation { continuation in
+            requestPasswordController = RequestPasswordController(
```

Comment:
do we need to retain RequestPasswordController ? as it is retained by the presentingViewController no?